frontend/src/components/files/Preview.vue

```diff
@@ -133,16 +133,32 @@ export default {
     }
   },
   async mounted () {
-    window.addEventListener('keyup', this.key)
+    window.addEventListener('keydown', this.key)
     this.$store.commit('setPreviewMode', true)
     this.listing = this.oldReq.items
+    this.$root.$on('preview-deleted', this.deleted)
     this.updatePreview()
   },
   beforeDestroy () {
```

Comment:
Use `this.$root.$off('preview-deleted', this.deleted)` to stop listening the event.